packages/core/test/actorLogic.test.ts

```diff
@@ -971,4 +972,41 @@ describe('composable actor logic', () => {
       }
     });
   });
+
+  it('higher-level logic should work', () => {
+    const logged: any[] = [];
+    function withLogging<T extends ActorLogic<any, any>>(actorLogic: T) {
+      const enhancedLogic: T = {
+        ...actorLogic,
+        transition: (state, event, actorCtx) => {
+          logged.push(event.type);
+          return actorLogic.transition(state, event, actorCtx);
+        }
+      };
+
+      return enhancedLogic;
+    }
+
+    const machine = createMachine({
+      initial: 'start',
+      states: {
+        start: {
+          on: { next: 'working' }
+        },
+        working: {
+          on: { more: 'done' }
+        },
+        done: {}
+      }
+    });
+
+    const actor = createActor(withLogging(machine)).start();
+
+    actor.send({ type: 'next' });
+    actor.send({ type: 'more' });
```

Comment:
In #4496, the actor breaks on calling `getPersistedSnaphot`. This is what should be tested in addition.

Reply:
As a quick/temporary fix, we could just `.bind` this method along with the other ones that we already bind:
https://github.com/statelyai/xstate/blob/eef14a2d558a00ad75558fb44432601608690890/packages/core/src/StateMachine.ts#L145-L148

Reply:
Just did both of these thing; tests pass

Reply:
Thanks a lot for all your hard (and high quality) work !